src/librustc_trans/mir/constant.rs

```diff
@@ -706,7 +704,11 @@ impl<'a, 'tcx> MirConstContext<'a, 'tcx> {
 
                 let base = match tr_lvalue.base {
                     Base::Value(llval) => {
-                        let align = type_of::align_of(self.ccx, ty);
+                        let align = if type_is_sized(self.ccx.tcx(), ty) {
```

Comment:
`Base::Value` always has a sized LLVM value, otherwise `consts::addr_of{,_mut}` below it wouldn't work.
*However*, I believe that what @arielb1 is pointing out is that `ty` could be unsized, and the alignment would be lost, even though the value itself corresponds to some sized value, e.g. the `simd_vec` in his example.

One possible fix is changing `consts::addr_of{,_mut}` to take an `Option` for alignment (and not change it if `None`).
That way, unsized values can just pass `None` and LLVM will just usean alignment based on the value.